endpoint/webhook/lekabapi/transformer.ts

```diff
@@ -0,0 +1,555 @@
+//session timeout in seconds, new session gets generated afterwards
+//disable by setting to 0
+const SESSION_TIMEOUT = 1800;
+
+const HIDE_USER_ID = true
+const HIDE_SESSION_ID = true
+const HIDE_FROM_CHANNEL = true
+//method used for hiding
+const HASH_ALGORITHM = "sha256"
+/**
+ * LEKAB LekabApi interfaces
+ */
+interface ILekabApiMessageBase {
+	"to":[
+			{
+				"channel":string;
+				"address":string;	
+			}
+		]
+}
+
+interface ILekabApiMediaMessage extends ILekabApiMessageBase {
+	"media_message": {
+		"url": string;
+	  }
+}
+
+interface ILekabApiTextMessage extends ILekabApiMessageBase {
+	"message": {
+		"text_message": {
+		  "text": string;
+		}
+	}
+}
+
+interface ILekabApiLocationMessage {
+	"location_message": {
+		"title": string;
+		"label": string;
+		"coordinates": {
+		  "latitude": number;
+		  "longitude": number;
+		}
+	  }
+}
+
+interface ILekabApiQuickReplyMessage extends ILekabApiMessageBase {
+	"choice_message": {
+		"text_message": {
+		  "text": "Text"
+		},
+		"choices": TLekabApiChoiceComponent[];
+	}
+}
+
+type TLekabApiChoiceComponent = ILekabApiChoiceTextComponent | ILekabApiChoiceCallComponent | ILekabApiChoiceUrlComponent;
+
+interface ILekabApiChoiceTextComponent {
+		"text_message": {
+		  "text": string;
+		},
+		"postback_data": string;
+	  
+}
+
+interface ILekabApiChoiceCallComponent {
+	"call_message": {
+		"title": string;
+		"phone_number": string;
+	  },
+	  "postback_data": string;
+}
+
+interface ILekabApiChoiceUrlComponent {
+	"type": "reply",
+    "postback_data": string;
+    "url_suffix": string;
+}
+
+interface ILekabApiTemplateMessage extends ILekabApiMessageBase {
+	"template_message": {
+		"lekabapimessage_template": {
+		  "template_id": string;
+		  "language_code": string;
+		}
+		"body_parameters": ILekabApiTemplateBodyComponent[];
+		"header_parameters": ILekabApiTemplateHeaderComponent[];
+		"button_parameters": ILekabApiTemplateButtonComponent[];
+	}
+}
+
+
+type TLekabApiTemplateComponent = ILekabApiTemplateHeaderComponent | ILekabApiTemplateBodyComponent | ILekabApiTemplateButtonComponent;
+
+interface ILekabApiTemplateHeaderComponent {
+	"type": "text";
+	"text": string;
+	"link": string;
+	"provider_name": string;
+	"filename": string;  
+}
+
+interface ILekabApiTemplateBodyComponent {
+	type: string;
+}
+
+interface ILekabApiTemplateButtonComponent {
+	"type": "reply",
+    "postback_data": string;
+    "url_suffix": string;
+}
+
+
+type TLekabApiContent = ILekabApiTextMessage | ILekabApiMediaMessage | ILekabApiTemplateMessage | ILekabApiLocationMessage | ILekabApiQuickReplyMessage | any;
+
+const convertWebchatContentToLekabApi = (output, sessionId: string, sessionStorage: any): TLekabApiContent[] => {
+
+		// create list for lekabapimessage content
+		let lekabApiContents: TLekabApiContent[] = [];
+
+		// check if default text was sent
+		if (output.text && !output.data?._cognigy?._default && !output.data.type) {
```

Comment:
It might make sense to move this section into an else if block at the end, since currently a payload with text and output.data.type defined but not location will not do anything

Reply:
Hi.

So if someone tries for example this input in the Option/Data field in a Say node 
`{
	"type": "some user defined types that does not exist"
}`
and the output, received by transformer will something like 

`{
	"userId": "607337ed6ed16d0682ee3a7d77e89099855f1063a45252a91932e7679435ba38",
	"entrypoint": "637cc49ea961f163e8bcaf50",
	"flowId": "d7b3ab92-3c72-4a19-b196-f43dd1e630c3",
	"flowName": "Lekab API features",
	"text": "Something",
	"data": {
		"type": "some user defined types that does not exist"
	}
}`
 then an error is logged as follows

`error Missing LekabApi compatible channel output! {}`.

Did I miss something?

-Teemu

Reply:
Hi.

There is only one message where you need to define type in the Options/Data field while using LekabAPI transformer. It  is  when the message type is location.
So would it be better to guide users by writing in the README.md that "DON'T DEFINE TYPE in the Options/Data field unless you are sending location type message"?

-Teemu

Reply:
Hi, 
the thing is that in an omni-channel Cognigy Agent, there might be output.data.type payloads used for other channels, so it would be important that the Transformer would still support those cases.
The best practice here is that the text is used as much as possible and if there is no text and no known data payload that the fallback message is used as the last option.

Reply:
I'll change it.